compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -846,20 +910,33 @@ impl Context {
                 let mut values = Vector::new();
                 for _ in 0..len {
                     for typ in element_types.as_ref() {
-                        values.push_back(self.array_get_value(typ, block_id, var_index)?);
+                        values.push_back(self.array_get_value(
+                            typ,
+                            block_id,
+                            var_index,
+                            slice_sizes,
+                        )?);
                     }
                 }
                 Ok(AcirValue::Array(values))
             }
-            Type::Slice(_) => {
-                // TODO(#2752): need SSA values here to fetch the len like we do for a Type::Array
-                // Update this to enable fetching slices from nested arrays
-                Err(InternalError::UnExpected {
-                    expected: "array".to_owned(),
-                    found: ssa_type.to_string(),
-                    call_stack: self.acir_context.get_call_stack(),
+            Type::Slice(element_types) => {
+                // It is not enough to execute this loop and simply pass the size from the parent definition.
+                // We need the internal sizes of each type in case of a nested slice.
+                let mut values = Vector::new();
+                let current_size = slice_sizes[0];
+                slice_sizes.drain(0..1);
```

Comment:
```suggestion
                let current_size = slice_sizes.pop_front();
```

Reply:
This is switched to a slice in the followup PR https://github.com/noir-lang/noir/pull/3258 so it looks a bit different now